language/stdlib/modules/Libra.move

```diff
@@ -480,38 +482,38 @@ module Libra {
     }
-    spec schema BurnAbortsIf<CoinType> {
-        preburn: Preburn<CoinType>;
-        aborts_if !spec_is_currency<CoinType>();
-        aborts_if len(preburn.requests) == 0;
-        aborts_if {
-            let i = spec_currency_info<CoinType>();
-            i.total_value < preburn.requests[0].value || i.preburn_value < preburn.requests[0].value
-        };
-    }
-    spec schema BurnEnsures<CoinType> {
-        preburn: Preburn<CoinType>;
-        ensures Vector::eq_pop_front(preburn.requests, old(preburn.requests));
-        ensures spec_currency_info<CoinType>().total_value
-                == old(spec_currency_info<CoinType>().total_value) - old(preburn.requests[0].value);
-        ensures spec_currency_info<CoinType>().preburn_value
-                == old(spec_currency_info<CoinType>().preburn_value) - old(preburn.requests[0].value);
-    }
+    // TODO move prover change
+    // spec schema BurnAbortsIf<CoinType> {
+    //     preburn: Preburn<CoinType>;
+    //     aborts_if !spec_is_currency<CoinType>();
+    //     aborts_if len(preburn.requests) == 0;
+    //     aborts_if {
+    //         let i = spec_currency_info<CoinType>();
+    //         i.total_value < preburn.requests[0].value || i.preburn_value < preburn.requests[0].value
+    //     };
+    // }
+    // spec schema BurnEnsures<CoinType> {
+    //     preburn: Preburn<CoinType>;
+    //     ensures Vector::eq_pop_front(preburn.requests, old(preburn.requests));
+    //     ensures spec_currency_info<CoinType>().total_value
+    //             == old(spec_currency_info<CoinType>().total_value) - old(preburn.requests[0].value);
+    //     ensures spec_currency_info<CoinType>().preburn_value
+    //             == old(spec_currency_info<CoinType>().preburn_value) - old(preburn.requests[0].value);
+    // }
 
     /// Cancels the burn request in the `Preburn` resource stored at `preburn_address` and
     /// return the coins to the caller.
-    /// If there are multiple preburn requests in progress for `CoinType` (i.e. in the
-    /// preburn queue), this will cancel the oldest one.
     /// This function can only be called by the holder of a
     /// `BurnCapability<CoinType>`, and will fail if the `Preburn<CoinType>` resource
-    /// at `preburn_address` does not contain any pending burn requests.
+    /// at `preburn_address` does not contain a pending burn request.
     public fun cancel_burn_with_capability<CoinType>(
         preburn_address: address,
         _capability: &BurnCapability<CoinType>
     ): Libra<CoinType> acquires CurrencyInfo, Preburn {
-        // destroy the coin at the head of the preburn queue
+        // destroy the coin in the preburn area
         let preburn = borrow_global_mut<Preburn<CoinType>>(preburn_address);
-        let coin = Vector::remove(&mut preburn.requests, 0);
+        let coin = withdraw_all<CoinType>(&mut preburn.request);
+        deposit(&mut preburn.request, zero<CoinType>());
```

Comment:
Delete? This is depositing a coin with value zero into a coin with value zero :).